src/ray/object_manager/common.cc

```diff
@@ -59,41 +56,30 @@ void PrintPlasmaObjectHeader(const PlasmaObjectHeader *header) {
                  << "metadata_size: " << header->metadata_size << "\n";
 }
 
-Status PlasmaObjectHeader::TryAcquireWriterMutex() {
-  // Try to acquire the lock, checking every 1s for the error bit.
-  struct timespec ts;
-  do {
-    if (has_error) {
-      return Status::IOError("channel closed");
-    }
-    clock_gettime(CLOCK_REALTIME, &ts);
-    ts.tv_sec += 1;
-  } while (pthread_mutex_timedlock(&wr_mut, &ts));
-
+Status PlasmaObjectHeader::TryToAcquireSemaphore(sem_t *sem) const {
+  RAY_CHECK_EQ(sem_wait(sem), 0);
```

Comment:
Hmm don't we need to check `has_error` before the semaphore too? In case the caller tries this multiple times after the error bit is set.